airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1043,6 +1053,38 @@ def create_concurrent_cursor_from_datetime_based_cursor(
             if evaluated_step:
                 step_length = parse_duration(evaluated_step)
 
+        clamping_strategy = NoClamping()
+        if datetime_based_cursor_model.clamping:
+            match datetime_based_cursor_model.clamping.target:
+                case Clamping.target.DAY:
+                    clamping_strategy = DayClampingStrategy()
+                    end_date_provider = ClampingEndProvider(
+                        DayClampingStrategy(is_ceiling=False),
+                        end_date_provider,
+                        granularity=timedelta(seconds=1),
```

Comment:
Note that all granularity in [this implementation](https://github.com/airbytehq/airbyte/blob/1f40c82e100b85573fc4b29280f2dcf89bae1842/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py#L823-L853) seems to have been `1 microseconds`. 

However, I think we should update this to take `datetime_based_cursor_model.granularity`. WDYT?

Reply:
I say it's using microseconds because this is what the implementation of pendulum `x.end_of("day")` does